test/gtest/ucp/ucp_test.h

```diff
@@ -266,4 +266,36 @@ std::ostream& operator<<(std::ostream& os, const ucp_test_param& test_param);
     UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, self,   "self") \
     UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, tcp,    "tcp")
 
+
+/**
+ * Auxiliary macro to instantiate the parameterized test case for all
+ * transport combinations with GPU memory awareness
+ *
+ * @param _test_case     Test case class, derived from ucp_test.
+ * @param _gpu_copy_tls  GPU memory copy TLs (cuda_copy/rocm_copy/etc)
+ * @param _gpu_ipc_tls  GPU IPC TLs (cuda_ipc/rocm_ipc/etc)
+ */
+#define _UCP_INSTANTIATE_TEST_CASE_GPU_AWARE(_test_case, _gpu_copy_tls, _gpu_ipc_tls) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, dcx,        "dc_x,"_gpu_copy_tls) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, ud,         "ud_v,"_gpu_copy_tls) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, udx,        "ud_x,"_gpu_copy_tls) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, rc,         "rc_v,"_gpu_copy_tls) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, rcx,        "rc_x,"_gpu_copy_tls) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, shm_ib,     "shm,ib,"_gpu_copy_tls) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, shm_ib_ipc, "shm,ib,"_gpu_ipc_tls","_gpu_copy_tls) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, ugni,       "ugni,"_gpu_copy_tls) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, tcp,        "tcp,"_gpu_copy_tls)
+
+
+/**
+ * Instantiate the parameterized test case for all transport combinations
+ * with CUDA memory awareness
+ *
+ * @param _test_case  Test case class, derived from ucp_test.
+ */
+#define UCP_INSTANTIATE_TEST_CASE_CUDA_AWARE(_test_case) \
+    _UCP_INSTANTIATE_TEST_CASE_GPU_AWARE(_test_case, "cuda_copy", "cuda_ipc"), \
```

Comment:
maybe pass just `cuda_copy, cuda_ipc` without quotes.  Then `_UCP_INSTANTIATE_TEST_CASE_GPU_AWARE` would use just one pair of `""` for ipc initiation, like:
`UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, shm_ib_ipc, "shm,ib,_gpu_ipc_tls,_gpu_copy_tls") `

Reply:
better leave as is, since we need to do `UCS_PP_MAKE_STRING` to make string from unquoted TLs